### ISSUE TITLE
[Bug]: Add Playlist fails with 'failed to add item to index'

### ISSUE THREAD

**Issue by drummingdemon** [tubearchivist/tubearchivist#605]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Synology

## Describe the bug

Trying to add a YouTube Playlist fails with 'failed to add item to index' error. Log attached from Synology Container Manager with DJANGO_DEBUG set to 'true'. Attached full log as .txt file as the character limit is reached.

## Steps To Reproduce

Try to add this playlist: https://www.youtube.com/playlist?list=PL8mG-RkN2uTw7PhlnAr4pZZz2QubIbujH

## Expected behavior

Playlist to be added to the subscriptions.
[log.txt](https://github.com/tubearchivist/tubearchivist/files/13400209/log.txt)

### Relevant log output

```shell
[1;33m[2023-11-18 12:53:32,419: WARNING/ForkPoolWorker-4] dab0ba8b-8965-49d4-a2b7-02634f59e2b0 return callback[0m
ValueError: failed to add item to index[0m
raise ValueError("failed to add item to index")
File "/app/home/src/es/connect.py", line 112, in put
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_, _ = ElasticWrap(self.es_path).put(self.json_data, refresh=True)
File "/app/home/src/index/generic.py", line 57, in upload_to_es
playlist_h.upload_to_es()
File "/app/home/src/download/subscriptions.py", line 196, in process_url_str
PlaylistSubscription().process_url_str([item])
File "/app/home/src/download/subscriptions.py", line 357, in subscribe_type
self.subscribe_type(item, expected_type=expected_type)
File "/app/home/src/download/subscriptions.py", line 347, in subscribe
SubscriptionHandler(url_str, task=self).subscribe(expected_type)
File "/app/home/tasks.py", line 356, in subscribe_to
^^^^^^^^^^^^^^^^^^^^^^^^^
return self.run(*args, **kwargs)
File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 760, in __protected_call__
^^^^^^^^^^^^^^^^^^^^
R = retval = fun(*args, **kwargs)
File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
Traceback (most recent call last):
[1;31m[2023-11-18 12:53:32,417: ERROR/ForkPoolWorker-4] Task subscribe_to[dab0ba8b-8965-49d4-a2b7-02634f59e2b0] raised unexpected: ValueError('failed to add item to index')
[1;33m[2023-11-18 12:53:32,415: WARNING/ForkPoolWorker-4] dab0ba8b-8965-49d4-a2b7-02634f59e2b0 Failed callback[0m
id': 'ntUIv5pVC4Q', 'title': 'The WAN Show : 4960X Preview, Google Glass Hacked, and Guest LOGAN - July 19, 2013', 'uploader': 'Linus Tech Tips', 'idx': 378, 'downloaded': False}, {'youtube_id': 'dm6aRONewxk', 'title': 'The WAN Show : Hemlis, Exoskeletons, Consoles in China, and Guest PAUL - July 12, 2013', 'uploader': 'Linus Tech Tips', 'idx': 379, 'downloaded': False}], 'playlist_subscribed': True}[0m
apan! Intel Skylake & Apple Watch Reviews - April 10, 2015', 'uploader': 'Linus Tech Tips', 'idx': 291, 'downloaded': False}, {'youtube_id': 'U7b2F_a3FKY', 'title': 'The WAN Show - Apple Watch Edition Benefits & Amazon Testing Delivery Drones', 'uploader': 'Linus Tech Tips', 'idx': 292, 'downloaded': False}, {'youtube_id': 'i25yk1S7Sa0', 'title': 'The WAN Show - Night-vision for Humans & YouTube Targeting eSports Events - Mar 27, 2015', 'uploader': 'Linus Tech Tips', 'idx': 293, 'downloaded': False}, {'youtube_id': 'EK_K7IFP7Zo', 'title': 'The WAN Show - Windows 10 NOT Free for Pirates & More R9 390X Rumours - Mar 20, 2015', 'uploader': 'Linus Tech Tips', 'idx': 294, 'downloaded': False}, {'youtube_id': '7d_FkKYMSfo', 'title': 'The WAN Show - CRAZY NVIDIA and Valve News! Titan X and More! - Mar 6, 2015', 'uploader': 'Linus Tech Tips', 'idx': 295, 'downloaded': False}, {'youtube_id': 'TOCDKbbp6Fs', 'title': 'The WAN Show - Leonard Nimoy Passes & YouTube has 1B viewers & no profit! - Feb 27, 2015', 'uploader': 'Linus Tech Tips', 'idx': 296, 'downloaded': False}, {'youtube_id': 'imPyit7zZY0', 'title': 'The WAN Show - Are Old Video Games ACTUALLY Better? Apple Might Make a Car??  - Feb 20, 2015', 'uploader': 'Linus Tech Tips', 'idx': 297, 'downloaded': False}, {'youtube_id': 'w5boVxmH_Yc', 'title': 'The WAN Show - Your TV Might Have its Own Ads?? & NVIDIA Disables Overclocking - Feb 13, 2015', 'uploader': 'Linus Tech Tips', 'idx': 298, 'downloaded': False}, {'youtube_id': 'Mp5r1Py3GxA', 'title': 'The WAN Show - Time Warner Cable 97% Profit on Internet?? Also Other Things - Feb 6, 2015', 'uploader': 'Linus Tech Tips', 'idx': 299, 'downloaded': False}, {'youtube_id': 'BMBNNQWO-qw', 'title': 'The WAN Show - Love FCC, H8 on NVIDIA & Comcast Edition - Jan 30, 2015', 'uploader': 'Linus Tech Tips', 'idx': 300, 'downloaded': False}, 
... 
<truncated, see log.txt>
...
[1;33m[2023-11-18 12:53:32,406: WARNING/ForkPoolWorker-4] {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"can't merge a non object mapping [playlist_entries] with an object mapping"}],"type":"illegal_argument_exception","reason":"can't merge a non object mapping [playlist_entries] with an object mapping"},"status":400}[0m
[1;33m[2023-11-18 12:53:28,092: WARNING/ForkPoolWorker-4] PL8mG-RkN2uTw7PhlnAr4pZZz2QubIbujH: get metadata from youtube[0m
[1;33m[2023-11-18 12:53:28,092: WARNING/ForkPoolWorker-4] {"_index":"ta_playlist","_id":"PL8mG-RkN2uTw7PhlnAr4pZZz2QubIbujH","found":false}[0m
[1;33m[2023-11-18 12:53:28,088: WARNING/ForkPoolWorker-4] PL8mG-RkN2uTw7PhlnAr4pZZz2QubIbujH: get metadata from es[0m
[1;33m[2023-11-18 12:53:28,037: WARNING/ForkPoolWorker-4] subscribe_to create callback[0m
[2023-11-18 12:53:28,035: INFO/MainProcess] Task subscribe_to[dab0ba8b-8965-49d4-a2b7-02634f59e2b0] received
https://www.youtube.com/playlist?list=PL8mG-RkN2uTw7PhlnAr4pZZz2QubIbujH
[pid: 32|app: 0|req: 1/1] 192.168.1.70 () {46 vars in 790 bytes} [Sat Nov 18 12:52:52 2023] GET /api/video/TQ6o0-z4umc/ => generated 429 bytes in 187 msecs (HTTP/1.1 404) 9 headers in 354 bytes (1 switches on core 0)
Not Found: /api/video/TQ6o0-z4umc/
{"_index":"ta_video","_id":"TQ6o0-z4umc","found":false}
[2023-11-18 12:52:24,074: INFO/MainProcess] celery@tubearchivist ready.
```

**Comment by bbilly1** [comment]
Thanks for reporting, I can reproduce this on a blank index, where this playlist is the first playlist to be indexed. Something wrong with the mapping.

Quick workaround is to index another playlist first, that will create the expected mapping, after that, you should be able to add the currently failing playlist.

**Comment by drummingdemon** [comment]
Awesome, thank you so much for getting back to me so quickly - it does work 👍

**Comment by bbilly1** [comment]
fix is shipped with v0.4.4, thanks for reporting.
